README.md

```diff
@@ -89,6 +89,31 @@ $  ansible-playbook playbook.yml -i inventory.ini
 ```
 This will install the application on the target VM and start the application service. You should then be able to access the application by navigating to http://*your_target_vm_ip*:5000
 
+## Running the App in a container using Docker
+
+It is a prerequisite to have a Docker Engine running before attempting to build the environments. You can do this by installing Docker Desktop (licenses are free for educational use even in a professional environment): https://docs.docker.com/get-docker/
+
+Once you have the Docker Engine running you can build the development and production images using the following commands.
+
+## Building the Docker image per environment
+```bash
+$  docker build --target development --tag todo-app:dev
+$  docker build --target production --tag todo-app:prod
+```
+
+Once the images are successfully built, you can run the development and production containers using the following commands.
+
+## Running the Development container (with flask and hot reload)
+```bash
+$  docker run --env-file ./.env -p 5000:5000 --mount "type=bind,source=$(pwd)/todo_app,target=/opt/todoapp/todo_app" todo-app:dev
+```
+[NOTE: For running the development environment you can also use the docker-compose.yml file, by running the `docker compose up` command.]
```

Comment:
Minor personal preference, but I tend to recommend `docker compose up --build`, particularly when specifying an image name, because I've fallen foul of being surprised when it uses an existing out-of-date image once too often!
E.g. without this, if you added a new package locally (e.g. `poetry add pymongo`) and then updated your code, and _then_ ran the docker compose command, you'd pick up the new code (because of the bind mount) but it wouldn't rebuild the image and thus wouldn't have the new package